### PR TITLE
Add a variant of insertGraph that fills values map.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23894 Add logging to inliner.
* #21939 Add CallStack class.
* #23620 Make inlineCallTo to take Function instead of Graph as the callee argument.
* **#23619 Add a variant of insertGraph that fills values map.**

Pull Request resolved: https://github.com/pytorch/pytorch/pull/23619

Differential Revision: [D16591721](https://our.internmc.facebook.com/intern/diff/D16591721)

### CODE REVIEW

**Comment by zdevito** [comment]
Is there a reason to provide a Node map and not a value map? In the past, I think the value map has been more useful.

**Comment by ZolotukhinM** [comment]
> Is there a reason to provide a Node map and not a value map? In the past, I think the value map has been more useful.

That's a good point. In the end, having Value<->Value and Node<->Node maps should be equivalent, so I should be able to switch to Value maps (also, we've already been creating it anyway).

File: torch/csrc/jit/ir.cpp [hunk 1]
```diff
@@ -1638,6 +1641,14 @@ std::vector<Value*> insertGraph(
   return outputs;
 }
 
+std::vector<Value*> insertGraph(
+    Graph& g,
+    Graph& callee,
+    ArrayRef<Value*> inputs) {
+  std::unordered_map<Node*, Node*> node_map;
```
Comment (apaszke) [inline torch/csrc/jit/ir.cpp]: Is initializing an empty map fast?

**Review by apaszke — DISMISSED** [review]
Why isn't this used anywhere?

**Comment by ZolotukhinM** [comment]
@apaszke, yes, I plan to use it in callstack implementation. Basically, I would need to update nodes that were inlined from another function - another way would be to pass a flag to the `insertGraph` and do it inside, but I think it's uglier.

As for the performance implications, I don't think this function is on a critical path and the addition was pretty small too. But anyways, I changed it to use value_map which we've had already, so now the effect on performance should be even smaller.

**Comment by apaszke** [comment]
Well we will end up inserting all cloned nodes into the map even when no one has asked for it, and that's a lot of allocations, but I guess we can leave it for now.

**Review by zdevito — APPROVED** [review]

**Pull request closed without merging**